src/java/htsjdk/samtools/SAMRecord.java

```diff
@@ -305,119 +308,202 @@ private boolean hasMateReferenceName() {
     }
 
     /**
-     * @return Reference name, or null if record has no reference.
+     * @return Reference name, or NO_ALIGNMENT_REFERENCE_NAME (-1) if the record has no reference name
      */
-    public String getReferenceName() {
-        return mReferenceName;
-    }
+    public String getReferenceName() { return mReferenceName; }
 
-    public void setReferenceName(final String value) {
-        /* String.intern() is surprisingly expensive, so avoid it by looking up in sequence dictionary if possible */
-        if (NO_ALIGNMENT_REFERENCE_NAME.equals(value)) {
+    /**
+     * Sets the reference name for this record. If the record has a valid SAMFileHeader and the reference
+     * name is present in the associated sequence dictionary, the record's reference index will also be
+     * updated with the corresponding sequence index.
```

Comment:
"If `referenceName` is `NO_ALIGNMENT_REFERENCE_NAME`, sets the reference index to `NO_ALIGNMENT_REFERENCE_INDEX`